internal/util/fetch/fetch.go

```diff
@@ -129,46 +170,51 @@ func cloneAndCopy(ctx context.Context, r *git.RepoSpec, dest string) error {
 // for versioning multiple kpt packages in a single repo independently. It
 // relies on the private clonerUsingGitExec function to try fetching different
 // refs.
-func ClonerUsingGitExec(ctx context.Context, repoSpec *git.RepoSpec) error {
+func (c *Cloner) ClonerUsingGitExec(ctx context.Context) error {
 	const op errors.Op = "fetch.ClonerUsingGitExec"
 
 	// Create a local representation of the upstream repo. This will initialize
 	// the cache for the specified repo uri if it isn't already there. It also
 	// fetches and caches all tag and branch refs from the upstream repo.
-	upstreamRepo, err := gitutil.NewGitUpstreamRepo(ctx, repoSpec.CloneSpec())
-	if err != nil {
-		return errors.E(op, errors.Git, errors.Repo(repoSpec.CloneSpec()), err)
+	upstreamRepo, exists := c.cachedRepo[*c.repoSpec]
+	if !exists {
+		newUpstreamRemp, err := gitutil.NewGitUpstreamRepo(ctx, c.repoSpec.CloneSpec(), gitutil.WithFetchedRefs(c.fetchedRefs))
```

Comment:
still trying to parse this. So we are creating a new repo cache because the repo doesn't exists but we are initializing with fetchedRefs (which meant we fetched the repo before ?) ?

Reply:
Yeah we don't need fetch refs here. It was left over from my debugging, updated now.